Gordon360/Services/NewsService.cs

```diff
@@ -153,9 +184,161 @@ public StudentNewsViewModel EditPosting(int newsID, StudentNews newData)
+        /// The base64 data will not include the first part of a base64 image ("data:image/...").
+        /// This is because this part is removed in every image before being submitted, and it
+        /// is readded in the frontend before being displayed.
+        /// 
+        /// This helper function does not perform any error checking; every place that calls it
+        /// checks that the path is not empty. Theoretically if it isn't empty it's certainly a valid path.
+        /// </summary>
+        /// <param name="imagePath">The path to the image</param>
+        /// <returns>The base64 content of the image</returns>
+        private string GetBase64ImageDataFromPath(string imagePath)
+        {
+            using (System.Drawing.Image image = System.Drawing.Image.FromFile(imagePath))
+            using (MemoryStream data = new MemoryStream())
+            {
+                image.Save(data, image.RawFormat);
+                byte[] imageBytes = data.ToArray();
+                string base64Data = System.Convert.ToBase64String(imageBytes);
+                return base64Data;
+            }
+        }
+
+        /// <summary>
+        /// Iterates through a collection of news items that have just been queried from
+        /// the database, replacing the Image entry of each (which contains the path to an image)
+        /// with the image itself. This is done because when the frontend makes requests for news items,
+        /// it needs the image data and not the path, but we store the path in the DB for efficiency and
+        /// storage reasons.
+        /// </summary>
+        /// <param name="items">The collection of student news entries</param>
+        /// <returns>The collection of entries with base64 data in their image variable</returns>
+        private IEnumerable<StudentNewsViewModel> InsertImageData(IEnumerable<StudentNewsViewModel> items)
+        {
+            foreach (StudentNewsViewModel item in items)
+            {
+                string imagePath = item.Image;
+
+                if (imagePath != null)
+                {
+                    item.Image = GetBase64ImageDataFromPath(imagePath);
+                }   
+            }
+            return items;
+        }
+
+        /// <summary>
+        /// Uploads a news image
+        /// Can be used to add an image to a new posting or to replace an image
+        /// for an existing posting.
+        /// 
+        /// Because editPosting calls this method to replace whatever is currently stored
+        /// (if anything) with new data, and because the new data can actually include
+        /// an image property of null (if the user removed the image from their post), it's
+        /// possible for 
+        /// </summary>
+        /// <param name="imageData">The base64 image data to be stored</param>
+        /// <param name="snid">The SNID of the news item to which the image belongs</param>
+        /// <returns>The status successful or failure</returns>
```

Comment:
Returning a string for success or failure is bad practice - in general and especially in C#. You should use exceptions instead, which are an improvement for error handling over strings. That way the calling code can programmatically handle different failure cases - plus you get s stack trace of what went wrong where, and you don't have to depend on typing/copying a string correctly to catch the right errors.

Reply:
Hmm. Every return string I wrote was actually not an error, with the exception of the one inside the catch; for that one, would it be good to use `System.Diagnostics.Debug.WriteLine("Something went wrong trying to save the image for submission with SNID " + snid)` or just remove it since the exception will also be logged?

Reply:
I would just remove it.